backend/services/inventory/api/http/api_inventory_test.go

```diff
@@ -54,49 +68,47 @@ func ToJson(data interface{}) string {
 	return string(j)
 }
 
-// test.HasHeader only tests the first header,
-// so create a wrapper for headers with multiple values
-func HasHeader(hdr, val string, r *test.Recorded) bool {
-	rec := r.Recorder
-	for _, v := range rec.Header()[hdr] {
-		if v == val {
-			return true
-		}
-	}
-
-	return false
-}
-
-func ExtractHeader(hdr, val string, r *test.Recorded) string {
-	rec := r.Recorder
-	for _, v := range rec.Header()[hdr] {
-		if v == val {
-			return v
-		}
-	}
-
-	return ""
-}
-
 func RestError(status string) map[string]interface{} {
 	return map[string]interface{}{"error": status, "request_id": "test"}
 }
 
 func runTestRequest(t *testing.T, handler http.Handler, req *http.Request, resp JSONResponseParams) {
 	req.Header.Add(requestid.RequestIdHeader, "test")
-	recorded := test.RunRequest(t, handler, req)
-	CheckRecordedResponse(t, recorded, resp)
+	w := httptest.NewRecorder()
+	handler.ServeHTTP(w, req)
+	CheckRecordedResponse(t, w, resp)
 }
 
 func makeMockApiHandler(t *testing.T, i inventory.InventoryApp) http.Handler {
-	handlers := NewInventoryApiHandlers(i)
-	assert.NotNil(t, handlers)
+	router := NewRouter(i)
+	assert.NotNil(t, router)
+	return router
+}
+
+func makeTestRequest(r *TestRequest) *http.Request {
+	var body io.Reader
+	if r.Body != nil {
+		bodyJSON, _ := json.Marshal(r.Body)
+		body = bytes.NewReader(bodyJSON)
+	}
 
-	handler, err := handlers.Build()
-	assert.NotNil(t, handler)
-	assert.NoError(t, err)
+	req, _ := http.NewRequest(r.Method, r.Path, body)
 
-	return handler
+	if r.Auth {
+		if r.Token == "" {
+			req.Header.Set("Authorization", DEFAULT_AUTH)
+		} else {
+			req.Header.Set("Authorization", r.Token)
+		}
+	}
+
+	if r.ContentType == "" {
+		req.Header.Set("Content-Type", "application/json")
+	} else {
+		req.Header.Set("Content-Type", r.ContentType)
+	}
```

Comment:
Technically `Content-Type` doesn't make sense if there's no content.
```suggestion
	if body != nil && r.ContentType == "" {
		req.Header.Set("Content-Type", "application/json")
	} else {
		req.Header.Set("Content-Type", r.ContentType)
	}
```